### PR TITLE
Referee thread unlock player

### PR DESCRIPTION
**Pull request by nickbar01234**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,10 +9,10 @@ install: requirements.txt
 	$(PYTHON) -m pip install -r requirements.txt
 
 client: install
-	.venv/bin/python3.11 src/client.py
+	.venv/bin/python3.12 src/client.py
```
Comment (nickbar01234) [inline Makefile:12]: Is this fine for you @tonyhuang12138?

File: src/client.py [hunk 2]
```diff
@@ -11,9 +11,9 @@ class Client:
     def __init__(self):
         self.network = Network()
         self.state = PlayerState(self.network)
-        self.network_barrier = threading.Semaphore(0)
+        self.player_start_barrier = threading.Semaphore(0)
```
Comment (ZiqiXu404) [inline src/client.py:14]: nice!

File: src/modules/state/server_state.py [hunk 3]
```diff
@@ -153,3 +146,9 @@ def get_questions(self):
     def set_questions(self, new_questions):
         with self.__questions_lock:
             self.__questions = new_questions
+
+    def wait_game_start(self):
```
Comment (ZiqiXu404) [inline src/modules/state/server_state.py:150]: nice abstraction

**Review by ZiqiXu404 — APPROVED** [review]
lg!!

File: Makefile (hunk 1 above)
Comment (tonyseetonydo) [reply Makefile:12]: I am still a little confused about how to correctly set this up. From what I could observe, running make install creates a .venv/bin directory with python3, and whatever the current python3 version you have. For me, that is python 3.9. So the executables I have are python3 and python3.9. This means that I cannot run the script with 3.12. 

Does this imply that we require the user to have installed python 3.12? If so, should we add this in the `Makefile`? Or should we add in `requirements.txt`

Comment (tonyseetonydo) [reply Makefile:12]: Nvm forget about `requirements.txt`, I realized they are python packages only.

File: src/modules/state/player_state.py [hunk 4]
```diff
@@ -32,6 +30,9 @@ def __init__(self, network):
         self.__game_state = GameState.START
         self.game_starts = threading.Semaphore(0)
 
+        # TODO(nickbar01234) - Should this state be here?
```
Comment (tonyseetonydo) [inline src/modules/state/player_state.py:33]: I think it makes sense to me? What's another place we can put it?

**Review by tonyseetonydo — APPROVED** [review]
Left some non-blocking comments but everything else looks good! Thanks for clean up + abstractions!

File: Makefile (hunk 1 above)
Comment (tonyseetonydo) [reply Makefile:12]: Discussed via chat and added to tickets.

**Pull request merged**